### PR TITLE
Add ".xyz" extension to PointSet.save

### LINKED ISSUE #2556: Unable to save PointSet to file
**Describe the bug, what's wrong, and what you expect:**

Creating a PointSet instance and trying to save it to a file fails.

-----

**To Reproduce**

```py
import numpy as np
import pyvista

rng = np.random.default_rng()
points = rng.random((10, 3))
pset = pyvista.PointSet(points)
pset.save("pset.vtk")
```

**Screenshots**

Not really a screenshot, but the result of this is:

```txt
ValueError: Invalid file extension for this data type. Must be one of: dict_keys([])
```

This suggests there are no supported formats (trying `ply` also doesn't work, for example).

-----

**System Information:**

```txt
--------------------------------------------------------------------------------
  Date: Tue May 03 13:53:48 2022 BST
                OS : Linux
            CPU(s) : 12
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : AMD
      GPU Renderer : AMD Radeon RX 5600 XT (NAVI10, DRM 3.44.0, 5.17.4-200.fc35.x86_64, LLVM 13.0.0)
       GPU Version : 4.6 (Core Profile) Mesa 21.3.8
  Python 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 11.2.1 20220127 (Red Hat
  11.2.1-9)]
           pyvista : 0.34.1
               vtk : 9.1.0
             numpy : 1.22.3
           imageio : 2.19.0
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.5.2
             scipy : 1.8.0
--------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

`PointSet.save` cannot be written because `_WRITER` does not exist, and an error always occurs. Therefore, in this PR, I have clarified which extension to specify by defining `_WRITER`. IMO, the only Writer object that can be used with `PointSet` is `vtkSimplePointsWriter`, which creates [xyz file](https://www.cgl.ucsf.edu/chimera/docs/UsersGuide/xyz.html). (If there are other Writers that can be added, please point them out.)
 

resolves #2556

### Details

```python
import numpy as np
import pyvista

rng = np.random.default_rng()
points = rng.random((10, 3))
pset = pyvista.PointSet(points)
pset.save("pset.xyz")
```

### CODE REVIEW

File: tests/test_pointset.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ def test_pointset(pointset):
     pointset.points = np.ones((10, 3))
     assert np.allclose(pointset.points, 1)
 
+    filename = str(tmpdir.mkdir("tmpdir").join(f'tmp.xyz'))
+    pointset.save("")
```
Comment (tkoyama010) [inline tests/test_pointset.py:38]:
```suggestion
    pointset.save("filename")
```

File: tests/test_pointset.py [hunk 2]
```diff
@@ -34,6 +34,9 @@ def test_pointset(pointset):
     pointset.points = np.ones((10, 3))
     assert np.allclose(pointset.points, 1)
 
+    filename = str(tmpdir.mkdir("tmpdir").join(f'tmp.xyz'))
```
Comment (tkoyama010) [inline tests/test_pointset.py:37]:
```suggestion
    filename = str(tmpdir.mkdir("tmpdir").join(f'{"tmp.xyz"}'))
```

**Comment by MatthewFlamm** [comment]
This is a good idea.  What about this writer? [vtkSimplePointsWriter](https://vtk.org/doc/nightly/html/classvtkSimplePointsWriter.html) 

It would also be nice to have a reader for xyz files and we could test the roundtrip.  [vtkSimplePointsReader](https://vtk.org/doc/nightly/html/classvtkSimplePointsReader.html)

**Comment by MatthewFlamm** [comment]
One problem I see with this format is that I don't think that any point_data will get saved.  Maybe the best we can do is writing it to vtp or the xml version?

**Comment by tkoyama010** [comment]
Thanks for the comment. As you say we can [casting PointSet to PolyData](https://docs.pyvista.org/api/core/_autosummary/pyvista.PointSet.cast_to_polydata.html). I will improve this PR.

File: tests/test_pointset.py [hunk 3]
```diff
@@ -20,7 +20,7 @@ def test_pointset_basic():
     assert 'PointSet' in repr(pset)
 
 
-def test_pointset(pointset):
+def test_pointset(pointset, tmpdir):
```
Comment (tkoyama010) [inline tests/test_pointset.py:23]:
```suggestion
def test_pointset(pointset):
```

**Comment by tkoyama010** [comment]
Anyway, it would be useful to add xyz file output.

**Review by akaszynski — APPROVED** [review]
This is good to merge as-is. We can add a specialized reader as a next step.

**Pull request merged**